tests/kubesonic/test_k8s_join_disjoin.py

```diff
@@ -0,0 +1,387 @@
+import logging
+import pytest
+import time
+
+from datetime import datetime
+from tests.common.helpers.assertions import pytest_assert
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('any'),
+    pytest.mark.disable_loganalyzer
+]
+
+MINIKUBE_VERSION = "v1.34.0"
+MINIKUBE_PATH = "/usr/local/bin/minikube"
+MINIKUBE_VIP = "control-plane.minikube.internal"
+MINIKUBE_DEFAULT_IP = "192.168.49.2"
+NO_PROXY = f"NO_PROXY={MINIKUBE_DEFAULT_IP}"
+MINIKUBE_SETUP_MAX_SECOND = 600
+MINIKUBE_SETUP_CHECK_INTERVAL = 10
+KUBERNETES_VERSION = "v1.22.2"
+KUBELET_CONFIGMAP = "kubelet-config-1.22"
+DAEMONSET_NODE_LABEL = "deployDaemonset"
+DAEMONSET_POD_LABEL = "test-ds-pod"
+DAEMONSET_CONTAINER_NAME = "mock-ds-container"
+VMHOST_PARAM_DEFAULT = None
+DUT_CERT_DIR = "/etc/sonic/credentials"
+DUT_CERT_BAK = f"{DUT_CERT_DIR}.bak"
+DUT_HOSTS_FILE = "/etc/hosts"
+DUT_PAUSE_IMAGE = "k8s.gcr.io/pause:3.5"
+
+
+def check_dut_k8s_version_supported(duthost):
+    logger.info("Check if the k8s version is supported")
+    k8s_version = duthost.shell("kubeadm version -o short", module_ignore_errors=True)["stdout"]
+    if k8s_version != KUBERNETES_VERSION:
+        logger.info(f"K8s version {k8s_version} is not supported")
+        exit(0)
+    logger.info(f"K8s version {k8s_version} is supported")
+
+
+def download_minikube(vmhost, creds):
+    logger.info("Start to download minikube")
+    minikube_url = f"https://github.com/kubernetes/minikube/releases/download/{MINIKUBE_VERSION}/minikube-linux-amd64"
+    tmp_location = "/tmp/minikube-linux-amd64"
+    http_proxy = creds.get("proxy_env", {}).get("http_proxy", "")
+    proxy_param = f"-x '{http_proxy}'" if http_proxy != "" else ""
+    vmhost.shell(f"curl -L {minikube_url} -o {tmp_location} {proxy_param}")
+    vmhost.shell(f"install {tmp_location} {MINIKUBE_PATH} && rm -f {tmp_location}")
+    logger.info("Minikube is downloaded")
+
+
+def remove_minikube(vmhost):
+    logger.info("Start to remove minikube")
+    vmhost.shell(f"rm -f {MINIKUBE_PATH}")
+    logger.info("Minikube is removed")
+
+
+def setup_k8s_master(vmhost, creds):
+    logger.info("Start to setup k8s master on vmhost")
+    http_proxy = creds.get("proxy_env", {}).get("http_proxy", "")
```

Comment:
why do we need http_proxy/https_proxy in this case? here we're talking to the vmhost so it should be all local access, right?

Reply:
During the Minikube cluster setup, it's going to download container images which is using to setup cluster. Still need the proxy.